core/dbt/adapters/base/connections.py

```diff
@@ -144,7 +145,7 @@ def set_connection_name(self, name: Optional[str] = None) -> Connection:
                 'Opening a new connection, currently in state {}'
                 .format(conn.state)
             )
-            conn.handle = LazyHandle(type(self))
+            conn.handle = LazyHandle(self.open)
```

Comment:
`LazyHandle` takes a direct reference to the method, to avoid some unnecessary weirdness with `abstractclassmethod`, inheritance, and mypy.